audplot/core/api.py

```diff
@@ -0,0 +1,179 @@
+import typing
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+
+import audmetric
+
+
+def confusion_matrix(
+        truth: typing.Union[typing.Sequence, pd.Series],
+        prediction: typing.Union[typing.Sequence, pd.Series],
+        *,
+        labels: typing.Sequence = None,
+        percentage: bool = True,
+):
+    r"""Confusion matrix between ground truth vs. predicted labels.
+
+    The confusion matrx is calculated by :mod:`audmetric.confusion_matrix`.
+
+    Args:
+        truth: truth values
+        prediction: predicted values
+        labels: labels to be included in confusion matrix
+        percentage: if ``True`` present the confusion matrix
+            with percentage values instead of absolute numbers
+
+    Example:
+        .. plot::
+            :context: reset
+            :include-source: false
+
+            import matplotlib.pyplot as plt
+            from audplot import confusion_matrix
+
+        .. plot::
+            :context: close-figs
+
+            >>> truth = ['A', 'A', 'A', 'B', 'B', 'B', 'C', 'C', 'C']
+            >>> prediction = ['A', 'A', 'B', 'B', 'C', 'C', 'A', 'A', 'C']
+            >>> confusion_matrix(truth, prediction, percentage=False)
+
+        .. plot::
+            :context: close-figs
+
+            >>> confusion_matrix(truth, prediction, labels=['A', 'B', 'C', 'D'])
+
+    """  # noqa: 501
+    sns.set()  # get prettier plots
+
+    labels = audmetric.core.utils.infer_labels(truth, prediction, labels)
```

Comment:
You think we should make `infer_labels` part of the official API?

Reply:
I was wondering about that as well. In the end I decided against it as under `audmetric` we only have metrics.
But we could maybe introduce `audmetric.utils` as this might indeed be helpful for others as well.